chemprop/models/model.py

```diff
@@ -169,15 +350,44 @@ def forward(self,
         :return: The output of the :class:`MoleculeModel`, containing a list of property predictions
         """
 
-        output = self.ffn(self.encoder(batch, features_batch, atom_descriptors_batch,
-                                       atom_features_batch, bond_features_batch))
+        output = self.ffn(
+            self.encoder(
+                batch,
+                features_batch,
+                atom_descriptors_batch,
+                atom_features_batch,
+                bond_features_batch,
+            )
+        )
 
         # Don't apply sigmoid during training b/c using BCEWithLogitsLoss
         if self.classification and not self.training:
             output = self.sigmoid(output)
         if self.multiclass:
-            output = output.reshape((output.size(0), -1, self.num_classes))  # batch size x num targets x num classes per target
+            output = output.reshape(
+                (output.size(0), -1, self.num_classes)
+            )  # batch size x num targets x num classes per target
             if not self.training:
-                output = self.multiclass_softmax(output)  # to get probabilities during evaluation, but not during training as we're using CrossEntropyLoss
+                output = self.multiclass_softmax(
+                    output
+                )  # to get probabilities during evaluation, but not during training as we're using CrossEntropyLoss
 
         return output
+
+
+class MoleculeModel(_MoleculeModel):
```

Comment:
@cjmcgill I think this what you're referring to? My naming is different, but I've already implemented a design like that

Reply:
Yep! Could have saved a lot by looking down to the bottom of your code. I thought it was just a conversion from implicit to explicit. Apologies.